### PR TITLE
docs(trainer): add note for NumPy randomness

### LINKED ISSUE #20527: Encourage NumPy dependent projects to upgrade to version `>=2.5.0`
### 📚 Documentation

As mentioned in the [Changelog](https://github.com/Lightning-AI/pytorch-lightning/pull/20369), version `2.5.0` introduces a robust scheme for seeding NumPy dependent dataloader workers.

As described in the related PR #20369, until this version Lightning was not handling correctly the seeding for NumPy, meaning that  all projects using NumPy for controlling randomness (e.g. random transformations for augmentation) are affected (see [here](https://www.reddit.com/r/MachineLearning/comments/mocpgj/p_using_pytorch_numpy_a_bug_that_plagues/) and also [here](https://tanelp.github.io/posts/a-bug-that-plagues-thousands-of-open-source-ml-projects/)).

Since NumPy is commonly used for augmentations in deep learning projects (see also the example from [PyTorch docs](https://pytorch.org/tutorials/beginner/data_loading_tutorial.html#transforms)) and proper control of seeding/randomness is essential for any project, I think it would be beneficial to add a note/warning in the docs to use version `2.5.0` or higher if randomness depends on NumPy.

cc @borda

### PR DESCRIPTION
**Pull request by adosar**
## What does this PR do?

Fixes #20527

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21310.org.readthedocs.build/en/21310/

### CODE REVIEW

File: docs/source-pytorch/common/trainer.rst [hunk 1]
```diff
@@ -165,6 +165,11 @@ By setting ``workers=True`` in :func:`~lightning.pytorch.seed_everything`, Light
 unique seeds across all dataloader workers and processes for :mod:`torch`, :mod:`numpy` and stdlib
 :mod:`random` number generators. When turned on, it ensures that e.g. data augmentations are not repeated across workers.
 
+.. note::
+
+    If your project depends on NumPy for randomness (e.g. for data
+    augmentation), it is recommended to use version 2.5.0 or higher.
+
```
Comment (deependujha) [inline docs/source-pytorch/common/trainer.rst:172]: shouldn't it be a note for `~lightning.pytorch.seed_everything`?

Why trainer?

Comment (adosar) [reply docs/source-pytorch/common/trainer.rst:172]: I added it in the `Trainer` since in this section reproducibility issues are discussed. Also, note that `seed_everything` is part of [Lightning Fabric](https://lightning.ai/docs/fabric/stable/api/fabric_methods.html#seed-everything).
